src/services/appointment.service.ts

```diff
@@ -0,0 +1,91 @@
+import { BadRequestException } from '@nestjs/common';
+import { Injectable } from '@nestjs/common';
+import { Repository, In } from 'typeorm';
+import { Appointment, User, Customer, Treatment } from '../entities/';
+import { AppointmentRegisterDto, CustomerRegisterDto } from '../dtos';
+import { InjectRepository } from '@nestjs/typeorm';
+import CustomerService from './customer.service';
+
+@Injectable()
+export default class AppointmentService {
+  constructor(
+    @InjectRepository(Appointment)
+    private readonly appointmentRepository: Repository<Appointment>,
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+    @InjectRepository(Customer)
+    private readonly customerRepository: Repository<Customer>,
+    @InjectRepository(Treatment)
+    private readonly treatmentRepository: Repository<Treatment>,
+    private readonly customerService: CustomerService,
+  ) {}
+
+  async getAllAppointments(): Promise<Appointment[]> {
+    return await this.appointmentRepository.find({
```

Comment:
I might be wrong, but since this method is already returning a promise you can omit the async-await:

```js
getAllAppointments(): Promise<Appointment[]> {
  return this.appointmentRepository.find({
    relations: { treatments: true },
  })
}
```